File: src-tauri/src/commands/get_java_version.rs
use std::{os::windows::process::CommandExt, process::Command};

use windows::Win32::System::Threading::CREATE_NO_WINDOW;

#[tauri::command]
pub fn get_java_version() -> Result<String, String> {
    let output = Command::new("javac")
        .arg("--version")
        .creation_flags(CREATE_NO_WINDOW.0)
        .output();

    match output {
        Ok(output) => {
            if output.status.success() {
                return Ok(String::from_utf8_lossy(&output.stdout).to_string());
            } else {
                return Err(String::from_utf8_lossy(&output.stderr).to_string());
            }
        }
        Err(error) => return Err(error.to_string()),
    }
}

File: src/lib/tauri/commands/get-java-version.ts
import { invoke } from "@tauri-apps/api";

export async function getJavaVersion() {
  try {
    return await invoke<string>("get_java_version");
  } catch (err) {
    switch (err) {
      case "program not found":
        return null;
      default:
        throw new Error("Failed to get Java version");
    }
  }
}
